js/foundation.offcanvas.js

```diff
@@ -118,10 +118,16 @@ class OffCanvas extends Plugin {
       }
     }
 
-    this.options.isRevealed = this.options.isRevealed || new RegExp(this.options.revealClass, 'g').test(this.$element[0].className);
+    // Get the revealOn option from the class.
+    var revealOnClass = this.$element[0].className.match(RegExpEscape(this.options.revealClass) + '([^\s])+', 'g');
+    if (revealOnClass) {
+      this.options.isRevealed = true;
+      this.options.revealOn = this.options.revealOn || revealOnClass[1];
```

Comment:
It has to be `revealOnClass[0]` – but this only works if a match has been found and otherwise throw an error...

Reply:
Not anymore since I switched to `regexp.exec(str)`